fendermint/app/src/app.rs

```diff
@@ -164,6 +169,8 @@ where
     state_hist_size: u64,
     /// Tracks the validator
     validators: ValidatorTracker<C>,
+
+    latest_consensus_params: Arc<RwLock<Option<TendermintConsensusParams>>>,
     /// The cometbft client
     client: C,
```

Comment:
Having `client` embedded here (and inside `ValidatorTracker`) creates a circular dependency, which is problematic regardless of the current issue (unless explicit orchestration is implemented). It’s better to remove it entirely.

Reply:
Yeah, I don't think circular dependency is an issue here, but it's not being used so it should be removed.